extensions/blocks/dialogue/components/time-stamp-control.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	BaseControl,
+	__experimentalNumberControl as NumberControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+function checkValidValue( val, max ) {
+	if ( val > max ) {
+		return val = max;
+	} else if ( val < 0 ) {
+		return val = 0;
+	}
+
+	return val;
```

Comment:
I'm not sure what's more performant, but this could be:
```suggestion
	return Math.max( 0, Math.min( val, max ) );
```

Reply:
sounds nice. commiting...